src/feasibility_form_nls.jl

```diff
@@ -220,13 +220,15 @@ function jtprod!(nlp :: FeasibilityFormNLS, xr :: AbstractVector, v :: AbstractV
   return jtv
 end
 
-function hess_structure(nlp :: FeasibilityFormNLS)
+function hess_structure!(nlp :: FeasibilityFormNLS, rows :: AbstractVector{<: Integer}, cols :: AbstractVector{<: Integer})
   n, m, ne = nlp.internal.meta.nvar, nlp.internal.meta.ncon, nlp.internal.nls_meta.nequ
-  IF, JF = hess_structure_residual(nlp.internal)
-  (Ic, Jc) = m > 0 ? hess_structure(nlp.internal) : (Int[], Int[])
-  I = [IF; Ic; (n+1:n+ne)]
-  J = [JF; Jc; (n+1:n+ne)]
-  return I, J
+  nnzhF = nlp.internal.nls_meta.nnzh
+  nnzhc = m > 0 ? nlp.internal.meta.nnzh : 0
+  I = 1:nnzhF
+  @views hess_structure_residual!(nlp.internal, rows[I], cols[I])
+  I = nnzhF+1:nnzhF+nnzhc
+  @views hess_structure!(nlp.internal, rows[I], cols[I])
```

Comment:
Here we should check for `m > 0` and add the new residual Hessian.
```
if m > 0                                                                                  
  I = nnzhF+1:nnzhF+nnzhc                                                                 
  @views hess_structure!(nlp.internal, rows[I], cols[I])                                  
end                                                                                       
I = nnzhF+nnzhc+1:nnzhF+nnzhc+ne                                                          
rows[I] .= n+1:n+ne                                                                       
cols[I] .= n+1:n+ne 
```

Reply:
I thought the check for `m > 0` was redundant. Wouldn't `nnzhc` be zero if `m` is zero?

Reply:
Because of line
```
nnzhc = m > 0 ? nlp.internal.meta.nnzh : 0
```
we have `nnzhc = 0` if `m == 0`, but `nlp.internal.meta.nnzh` is not necessarily 0, because of the objective function of `nlp.internal`. This means that `hess_structure!` on `nlp.internal` will try to return `rows` and `cols` using `nlp.internal.meta.nnzh`, and in this case it will break.